tests/system.rs

```diff
@@ -485,3 +503,142 @@ fn string_eq_ignore_newline_repr_test() {
+                    value: config_value.unwrap(),
+                },
+                code_block: code_block.join("\n"),
+                code_block_start_line,
+            });
+
+            config_value = None;
+            code_block.clear();
+        } else if line.starts_with("```rust") {
+            // We've reached the beginning of a Rust code block.
+            extracting_block = true;
+
+            // +1 to translate to one-based indexing
+            // +1 to get to first line of code (line after "```")
+            code_block_start_line = (i + 2) as u32;
+
+            // Assert that we have a configuration name and value.
+            assert!(
+                config_name.is_some(),
+                "configuration name not found for block beginning at line {}",
+                code_block_start_line
+            );
+            assert!(
+                config_value.is_some(),
+                "configuration value not found for block beginning at line {}",
+                code_block_start_line
+            );
+        } else if extracting_block {
+            code_block.push(line);
+        } else if let Some(c) = config_name_regex.captures(&line) {
+            config_name = Some(String::from(&c[1]));
+        } else if let Some(c) = config_value_regex.captures(&line) {
+            config_value = Some(String::from(&c[1]));
+        }
+    }
+
+    code_blocks
+}
+
+// Format the code blocks extracted in `get_code_blocks` and check for
+// failures.
+fn check_blocks_idempotency(blocks: &Vec<CodeBlock>) {
+    let mut failures = 0;
+
+    for block in blocks {
+        let input = Input::Text(block.code_block.clone());
+        let mut config = Config::default();
+        config.override_value(&block.config.name, &block.config.value);
+
+        let (error_summary, file_map, _report) =
+            format_input::<io::Stdout>(input, &config, None).unwrap();
+
+        if error_summary.has_parsing_errors() {
+            write_message(String::from(format!(
+                "\u{261d}\u{1f3fd} Failed to format block starting at Line {} in Configurations.md",
+                block.code_block_start_line
+            )));
+            failures += 1;
+        } else {
+            for &(ref _file_name, ref text) in &file_map {
+                let compare = make_diff(&block.code_block, text, DIFF_CONTEXT_SIZE);
+                if !compare.is_empty() {
+                    let mut mismatches = HashMap::new();
+                    mismatches.insert(PathBuf::from("Configurations.md"), compare);
+                    print_mismatches(mismatches, |line_num| {
+                        format!(
+                            "\nMismatch at Configurations.md:{}:",
+                            line_num + block.code_block_start_line - 1
+                        )
+                    });
+                    failures += 1;
+                }
+            }
+        }
+    }
+
+    // Display results.
```

Comment:
Could you move the println and assertion to the caller please? That feels like it makes this a more self-contained function.